lib/util/grpc.ex

```diff
@@ -0,0 +1,84 @@
+defmodule Util.Grpc do
+  @moduledoc """
+  This module provides an opiniated interface for communicating with
+  GRPC services. It handles starting and closing connections, logging,
+  metrics, and proper error handling.
+
+  Usage:
+
+  1. First, add Util.Grpc to your application's supervision tree.
+     This process keeps the configuration values for your Grpc clients.
+
+     children = [
+       worker(Util.Grpc, [])
+     ]
+
+     Supervisor.start_link(children, opts)
+
+  2. Describe the outgoing connections from your service:
+
+     Util.Grpc.setup([
+       Util.Grpc.Client.new(:user_service, "localhost:50051", UserApi.Stub),
+       Util.Grpc.Client.new(:billing_service, "localhost:50051", BillingApi.Stub)
+     ])
+
+  3. User Grpc.do to communicate with your upstream services:
+
+    req = ExampleApi.DescribeRequest.new(name: "a")
+
+    {:ok, res} = Util.Grpc.call(:user_service, :describe, req)
+
+  During the execution of the call, the following metrics are published:
+
+    - gprc.<client_name>.<method_name>.duration
```

Comment:
Can we add a tag with the source service? So we will know from where we are getting these metrics?

Reply:
Every Watchman metric already contains this info. The tag name is `service`.

Reply:
🤯 🙇